core/runtime/module.hpp

```diff
@@ -29,9 +29,35 @@ namespace kagome::runtime {
         const = 0;
   };
 
-  struct SingleModuleCache {
-    SingleModuleCache() : module{std::nullopt} {};
-    std::optional<std::shared_ptr<Module>> module;
+  /**
+   * A wrapper for compiled module. Used when we update WAVM runtime in order to
+   * skip double compilation which takes significant time (see issue #1104).
+   * Currently it's shared through DI.
+   */
+  class SingleModuleCache {
+   public:
+    SingleModuleCache() : module_{std::nullopt} {};  // Removes copy ctor
```

Comment:
You can actually just remove copy constructor explicitly if you want to highlight this fact.